### PR TITLE
Consumption RP - Introducing API Version 2019-10-01 with support of new Credits, lots, Event APIs, and Modern scopes for Usage Details ,and Charges APIs.

### PR DESCRIPTION
**Pull request by parisa-naeimi**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/parisa-naeimi/azure-rest-api-specs/dev-consumption-Microsoft.Consumption-2019-10-01...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by parisa-naeimi** [comment]
Hi @erich-wang  could you please check? Also can you please ask one of ARM team member check this PR? thank you

**Review by KrisBash — APPROVED** [review]
Signing off with comment. Please do note the comments and confirm correctness

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json [hunk 1]
```diff
@@ -3232,6 +3422,224 @@
...
+		"expirationDate": {
+		  "description": "Expiration Date.",
+		  "type": "string",
+		  "format": "date-time",
+		  "readOnly": true
+		},
+		"poNumber": {
+		  "description": "PO Number.",
+		  "type": "string",
+		  "format": "string",
```
[the remark this answers is not in the document]
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json]: @KrisBash Thank you for reviewing this PR, I think I already removed this. this is from a previous commit.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json [hunk 2]
```diff
@@ -3232,6 +3422,224 @@
...
+		"transactionDate": {
+			"description": "Transaction Date.",
+			"type": "string",
+			"format": "date-time",
+			"readOnly": true
+		},
+		"description": {
+			"description": "Transaction description.",
+			"type": "string",
+			"format": "string",
```
[the remark this answers is not in the document]
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json]: @KrisBash same as the other one, this is from an early commit.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json [hunk 3]
```diff
@@ -490,13 +517,34 @@
           {
             "$ref": "#/parameters/apiVersionParameter"
           },
+		  {
+			"name": "startDate",
```
[the remark this answers is not in the document]
Comment (parisa-naeimi) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2019-10-01/consumption.json]: This is actually string, 2019-10-01 it is not in date-time format.

**Comment by parisa-naeimi** [comment]
@erich-wang could you please check this PR?

**Pull request merged**